### PR TITLE
PHDO-249 Enable local integration of phdo components

### PR DESCRIPTION
**Pull request by whytheplatypus**
- Allow setting mapped port to avoid 8080 collision
- Set custom aws endpoint though env
- ~decode base64 sqs message body~

### CODE REVIEW

File: pstatus-report-sink-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/plugins/AWSSQS.kt [hunk 1]
```diff
@@ -2,159 +2,197 @@ package gov.cdc.ocio.processingstatusapi.plugins
...
-            secretAccessKey = this@AWSSQServiceConfiguration.secretAccessKey
-        }; region = this@AWSSQServiceConfiguration.region }
+    fun createSQSClient(): SqsClient {
+        return SqsClient {
+            credentialsProvider = StaticCredentialsProvider {
+                accessKeyId = this@AWSSQServiceConfiguration.accessKeyID
+                secretAccessKey = this@AWSSQServiceConfiguration.secretAccessKey
+            }
+            region = this@AWSSQServiceConfiguration.region
+            endpointUrl = this@AWSSQServiceConfiguration.endpoint
```
Comment (mkrystof) [inline pstatus-report-sink-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/plugins/AWSSQS.kt:45]: I don't see that `endpointUrl` is used anywhere.  Did you forget to use it?

File: docker-compose.yml [hunk 2]
```diff
@@ -19,7 +19,7 @@ services:
   report-sink:
     image: quay.io/us-cdcgov/phdo/pstatus-report-sink:latest
     ports:
-      - "127.0.0.1:8080:8080" # Rest API listener port
+      - "127.0.0.1:${SINK_PORT:-8080}:8080" # Rest API listener port
```
Comment (mkrystof) [inline docker-compose.yml:22]: Would you mind updating the main repo readme to indicate the `SINK_PORT` env var needs to be set?  Maybe with a suggestion.  Or, is there a way to default it to 8080?

File: pstatus-report-sink-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/plugins/AWSSQS.kt [hunk 3]
```diff
@@ -2,159 +2,197 @@ package gov.cdc.ocio.processingstatusapi.plugins
...
-                        val receiveMessageRequest = ReceiveMessageRequest {
-                            this.queueUrl = queueUrl
-                            maxNumberOfMessages = 5
+                    receivedMessages.messages?.forEach { message ->
+                        SchemaValidation.logger.info(
+                                "Received message from AWS SQS: ${message.body}"
+                        )
+                        val awsSQSProcessor = AWSSQSProcessor()
+                        message.body?.let {
+                            val rawMessage = components.gson.fromJson(it, SQSMessage::class.java)
```
Comment (mkrystof) [inline pstatus-report-sink-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/plugins/AWSSQS.kt:120]: Is there anyway from the message headers or metadata to determine if the message is raw or base64 encoded?  If so, would love to support either.  If everything needs to be base64 encoded then we need to update our load testing script to encode all the reports it sends.  Not a big deal, but would love to support either and figure out which one it is programmatically at runtime.

File: pstatus-report-sink-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/plugins/AWSSQS.kt [hunk 4]
```diff
@@ -2,159 +2,197 @@ package gov.cdc.ocio.processingstatusapi.plugins
...
-        CoroutineScope(Dispatchers.IO).launch {
-            while (true) {
-                var receivedMessages: ReceiveMessageResponse?
+            /**
+             * Validates messages from the AWS SQS Service
+             * @param receivedMessages the list of message(s) received from the queue to be
+             * validated
+             * @throws Exception thrown during validation `and` it's important to delete the message
+             * as it will be persisted to dead-letter container
+             * ```
```
Comment (mkrystof) [inline pstatus-report-sink-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/plugins/AWSSQS.kt:108]: Looks like you ran an auto-reformat on the code and it something funky with this comment formatting.  See the three ticks?

**Review by mkrystof — CHANGES_REQUESTED** [review]

File: pstatus-report-sink-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/plugins/AWSSQS.kt (hunk 3 above)
Comment (whytheplatypus) [reply pstatus-report-sink-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/plugins/AWSSQS.kt:120]: good catch, i'm going to make this unnecessary, but i do think there's a content type attribute for messages

File: docker-compose.yml (hunk 2 above)
Comment (whytheplatypus) [reply docker-compose.yml:22]: yup this defaults it to 8080, I will update the readme as well though

File: pstatus-report-sink-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/plugins/AWSSQS.kt (hunk 4 above)
Comment (whytheplatypus) [reply pstatus-report-sink-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/plugins/AWSSQS.kt:108]: I'm going to try and get rid of the formatting changes, sry editor did it without me noticing

File: pstatus-report-sink-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/plugins/AWSSQS.kt (hunk 1 above)
Comment (mkrystof) [reply pstatus-report-sink-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/plugins/AWSSQS.kt:43]: Oh I see now, thanks!

**Pull request merged**